lightning/src/ln/functional_test_utils.rs

```diff
@@ -1600,7 +1603,18 @@ pub fn do_claim_payment_along_route<'a, 'b, 'c>(origin_node: &Node<'a, 'b, 'c>,
 			assert_eq!(expected_next_node, node.node.get_our_node_id());
 			let update_next_msgs = !skip_last || idx != expected_route.len() - 1;
 			if next_msgs.is_some() {
-				mid_update_fulfill_dance!(node, prev_node, update_next_msgs);
+				// Since we are traversing in reverse, next_node is actually the previous node
+				let next_node: &Node;
+				if idx == expected_route.len() - 1 {
+					next_node = origin_node;
+				} else {
+					next_node = expected_route[idx - 1];
+				}
+				let next_channel_id = match node.node.list_usable_channels().iter().find(|&x| x.counterparty.node_id == next_node.node.get_our_node_id()) {
+					Some(channel) => channel.channel_id,
+					None => panic!("Uh oh! Coudn't find the channel")
+				};
```

Comment:
@TheBlueMatt Do we need to handle the case where there could be multiple channels opened within the same 2 nodes?

Reply:
> Do we need to handle the case where there could be multiple channels opened within the same 2 nodes?

Yea, maybe lets make the check "is the channel listed in the event one of the ones between the two given nodes".

> Also there are cases where the channel doesn't exist (eg channel_monitor_network_test

I assume those are cases when the channel is already closed? Can we detect that based on `fee_earned_msat` being `None` in the event?

Reply:
Cool, done.